pkg/crds/crds.go

```diff
@@ -0,0 +1,205 @@
+// Package crds is used for installing rancher CRDs.
+package crds
+
+import (
+	"context"
+	"embed"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/rancher/rancher/pkg/fleet"
+	"github.com/rancher/wrangler/pkg/crd"
+	"github.com/rancher/wrangler/pkg/yaml"
+	apiextv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/selection"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	// yamlDir directory holding CRDs yaml files.
+	yamlDir = "yamls"
+
+	// readyDuration time to wait for CRDs to be ready.
+	readyDuration = time.Minute * 1
+
+	k8sManagedByKey = "app.kubernetes.io/managed-by"
```

Comment:
Nit: This annotation (source: https://kubernetes.io/docs/concepts/overview/working-with-objects/common-labels/) seems to be used to indicate what tool is managing a resource (like helm). I would propose that we use `app.kubernetes.io/part-of` instead, since these definitions are part of the rancher app, but are technically managed/created by client-go.